### PR TITLE
community[patch]: speed up import times in the community package

### PR DESCRIPTION
**Pull request by leo-gan**
This PR speeds up import times in the community package

### CODE REVIEW

File: libs/community/langchain_community/chat_loaders/__init__.py [hunk 1]
```diff
@@ -17,3 +17,34 @@
...
+    "FolderFacebookMessengerChatLoader": "langchain_community.chat_loaders.facebook_messenger",  # noqa: E501
+    "GMailLoader": "langchain_community.chat_loaders.gmail",
+    "IMessageChatLoader": "langchain_community.chat_loaders.imessage",
+    "LangSmithDatasetChatLoader": "langchain_community.chat_loaders.langsmith",
+    "LangSmithRunChatLoader": "langchain_community.chat_loaders.langsmith",
+    "SingleFileFacebookMessengerChatLoader": "langchain_community.chat_loaders.facebook_messenger",  # noqa: E501
+    "SlackChatLoader": "langchain_community.chat_loaders.slack",
+    "TelegramChatLoader": "langchain_community.chat_loaders.telegram",
+    "WhatsAppChatLoader": "langchain_community.chat_loaders.whatsapp",
+    "map_ai_messages": "langchain_community.chat_loaders.utils",
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_loaders/__init__.py:35]:
```suggestion
```

File: libs/community/langchain_community/chat_loaders/__init__.py [hunk 2]
```diff
@@ -17,3 +17,34 @@
...
+    "map_ai_messages_in_session": "langchain_community.chat_loaders.utils",
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_loaders/__init__.py:36]:
```suggestion
```

File: libs/community/langchain_community/chat_loaders/__init__.py [hunk 3]
```diff
@@ -17,3 +17,34 @@
...
+    "merge_chat_runs": "langchain_community.chat_loaders.utils",
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_loaders/__init__.py:37]:
```suggestion
```

File: libs/community/langchain_community/chat_loaders/__init__.py [hunk 4]
```diff
@@ -17,3 +17,34 @@
...
+    "merge_chat_runs_in_session": "langchain_community.chat_loaders.utils",
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_loaders/__init__.py:38]:
```suggestion
```

File: libs/community/langchain_community/chat_loaders/__init__.py [hunk 5]
```diff
@@ -17,3 +17,34 @@
...
+    "nanoseconds_from_2001_to_datetime": "langchain_community.chat_loaders.imessage",
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_loaders/__init__.py:39]:
```suggestion
```

Comment (eyurtsev) [reply libs/community/langchain_community/chat_loaders/__init__.py:39]: looks like a somewhat strange functionality for a chat loader namespace to declare as a public function. feels like internal functionality

**Review by eyurtsev — COMMENTED** [review]
Good to merge if you're OK removing the extra imports.

It's hard to tell whether those are meant to be publicly exposed or not. We can revisit in a separate PR -- i would need more time to review though since I haven't worked with chat loaders much.

File: libs/community/langchain_community/chat_loaders/__init__.py (hunk 1 above)
Comment (leo-gan) [reply libs/community/langchain_community/chat_loaders/__init__.py:35]: @eyurtsev Do you want to hide only this function or all functions in `utils`?

**Comment by eyurtsev** [comment]
Thanks @leo-gan

**Comment by leo-gan** [comment]
@eyurtsev Thanks!

**Comment by pierreveron** [comment]
Hey @leo-gan and @eyurtsev, this PR removes the ability to have linting on the imports on VSCode (see image below). Using lazy loading with `__get_attr__ `seems not to be well supported (there's an old open [PR](https://github.com/pylint-dev/pylint/issues/4300) on Pylint about this). Do you have a solution to provide?

image

**Comment by eyurtsev** [comment]
could you check if this works for you: https://github.com/langchain-ai/langchain/pull/20050

**Pull request merged**